codegen/src/lintdoc.rs

```diff
@@ -1979,3 +2025,131 @@ fn extract_summary_from_rule(content: &str) -> String {
 
     events_to_text(events)
 }
+
+/// Parses markdown documentation to extract in-memory file systems scoped by section.
+/// To match the way we do rule checking in the Biome repo, we scope each
+/// file system to a markdown section. This allows lint rules to access
+/// multiple related files within the same documentation section while
+/// keeping different examples isolated.
+fn parse_file_system(docs: &'static str) -> Result<HashMap<usize, HashMap<String, String>>> {
```

Comment:
It's not very clear what a "section" is. You might want to clarify that.

Also, it would be great if we explain the business logic of the parsing somewhere

Reply:
I tried to improve the documentation to be more clear. Let me know if you think it's better or could be improved more